app/controllers/interviews_controller.rb

```diff
@@ -0,0 +1,177 @@
+        @interview.users << @candidate
+    else
+        @interview.users << User.new(candidate_params)
+    end
+
+    respond_to do |format|
+      if @interview.update(:interview_date => interview_params[:interview_date], :start_time => interview_params[:start_time], :end_time => interview_params[:end_time])
+        interview_update_mail
+        @entryx = @interview.interviews_users.where(:role => 0).first
+        @entryx.destroy
+        @currx = @interview.interviews_users.where(:role => nil).first
+        @currx.update(:role => 0)
+
+        @entryy = @interview.interviews_users.where(:role => 1).first
+        @entryy.destroy
+        @curry = @interview.interviews_users.where(:role => nil).first
+        @curry.update(:role => 1)
+        
+        format.html { redirect_to @interview, notice: 'Interview schedule was successfully updated.' }
+        format.json { render :show, status: :ok, location: @interview }
+      else
+        format.html { render :edit }
+        format.json { render json: @interview.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  def destroy
+    @interview.destroy
+    respond_to do |format|
+      format.html { redirect_to interviews_url, notice: 'Interview schedule was successfully destroyed.' }
+      format.json { head :no_content }
+    end
+  end
+
+  private
+    def set_interview
+        @interview = Interview.find(params[:id])
+    end
+
+    def interview_params
+        params.require(:interview).permit({interviews_users_attributes: [users: [:name, :email, :resume]]}, :interview_date, :start_time, :end_time)
+    end
+
+    def welcome
+        candidate_params = interview_params[:interviews_users_attributes]["1"][:users]
+
+        h = JSON.generate({ 'type' => 'welcome', 
+            'name' => candidate_params[:name],
+            'email' => candidate_params[:email],
+            'message' => "Welcome to interview scheduler!"})
+    
+        PostmanWorker.perform_async(h, 1)
+    end
+
+    def interview_create_mail
+        candidate_params = interview_params[:interviews_users_attributes]["1"][:users]
+
+        h = JSON.generate({ 'type' => 'interview_create', 
+            'name' => candidate_params[:name],
+            'email' => candidate_params[:email],
+            'date' => interview_params[:interview_date],
+            'start_time' => interview_params[:start_time],
+            'end_time' => interview_params[:end_time]})
+    
+        PostmanWorker.perform_async(h, 1)
+    end
+
+    def interview_update_mail
+        candidate_params = interview_params[:interviews_users_attributes]["1"][:users]
+
+        h = JSON.generate({ 'type' => 'interview_update', 
+            'name' => candidate_params[:name],
+            'email' => candidate_params[:email],
+            'date' => interview_params[:interview_date],
+            'start_time' => interview_params[:start_time],
+            'end_time' => interview_params[:end_time]})
```

Comment:
If you're sending so much data to your job worker, you're doing it wrong. Pass only the interview id and type. Rest should be done within the worker